geode-dunit/src/distributedTest/java/org/apache/geode/test/dunit/rules/tests/SharedCountersRuleDistributedTest.java

```diff
@@ -163,17 +221,35 @@ private void givenExecutorInEveryVM(final int numThreads) {
     }
   }
 
-  private void submitIncrementTasks(int numThreads, final Serializable id) {
+  private void submitIncrementTasks(final int numThreads, final Serializable id) {
     for (int i = 0; i < numThreads; i++) {
       futures.add(supplyAsync(() -> {
         sharedCountersRule.increment(id);
         return true;
       }, executor));
     }
-    combined = CompletableFuture.allOf(futures.toArray(new CompletableFuture[futures.size()]));
+    combined = CompletableFuture.allOf(futures.toArray(new CompletableFuture[0]));
   }
 
   private static long calculateTimeoutMillis(final Stopwatch stopwatch) {
     return TWO_MINUTES_MILLIS - stopwatch.elapsed(MILLISECONDS);
   }
+
+  /**
+   * Used by test {@link #withIdInitializesIdInEveryVm()}
+   */
+  public static class WithIdInitializesIdInEveryVm implements Serializable {
```

Comment:
This inner class is focused on constructing the rule via a builder. Consider naming it something like 'SharedCountersRuleBuiltByBuilder`.